src/gradualizer_lib.erl

```diff
@@ -220,6 +222,19 @@ pick_value(Type, Env)
             throw({undef, Kind, Anno, {Name, length(Args)}})
     end.
 
+pick_binary_value(?type(binary, []), _Env) ->
+    {bin, erl_anno:new(0), []};
+pick_binary_value(?type(binary, [{integer, _, 0}, {integer, _, _}]), _Env) ->
+    {bin, erl_anno:new(0), []};
+pick_binary_value(?type(binary, [{integer, _, M}, {integer, _, _}]), _Env) when M > 0 ->
+    Elements = case M rem 8 of
+                   0 ->
+                       S = [ $a || _ <- lists:seq(1, M div 8) ],
```

Comment:
Good to simplify. Is there a need for `<<"aaaa...">>` or can we just use `<<0:M>>` here too? Is there is a nonempty string type in Elixir that would benefit from picking a string-looking value?

I think the minimal binary-picking would be just two clauses and no helper function needed, but it might have missed something.

```Erlang
pick_value(?type(binary, [{integer, _, M}, {integer, _, _}]), _Env) when M > 0 ->
    {bin, erl_anno:new(0), [{bin_element, 0, {integer, 0, 0}, {integer, 0, M}, default}]};
pick_value(?type(binary, _), _Env) ->
    {bin, erl_anno:new(0), []};
```

Reply:
Btw, I'd use `rem` in a guard, so the case clauses can be turned into function clauses, i.e. `when M > 0, M rem 8 =:= 0`.

Reply:
Oh, you removed this. :-) Well, I'll leave it to you to decide if you want the `<<"aaa...">>` back or not. I don't mind, just want to understand the use cases.